bgpd/bgp_attr.c

```diff
@@ -614,6 +618,91 @@ static void evpn_overlay_finish(void)
 			    (void (*)(void *))evpn_overlay_free);
 }
 
+static void *bgp_nhc_hash_alloc(void *p)
+{
+	return p;
+}
+
+static struct bgp_nhc *bgp_nhc_intern(struct bgp_nhc *nhc)
+{
+	struct bgp_nhc *find;
+
+	find = hash_get(bgp_nhc_hash, nhc, bgp_nhc_hash_alloc);
+	if (find != nhc)
+		bgp_nhc_free(nhc);
+	find->refcnt++;
+
+	return find;
+}
+
+static void bgp_nhc_unintern(struct bgp_nhc **nhcp)
+{
+	struct bgp_nhc *nhc = *nhcp;
+
+	if (!*nhcp)
+		return;
+
+	if (nhc->refcnt)
+		nhc->refcnt--;
+
+	if (nhc->refcnt == 0) {
+		hash_release(bgp_nhc_hash, nhc);
+		bgp_nhc_free(nhc);
+		*nhcp = NULL;
+	}
+}
+
+static uint32_t bgp_nhc_hash_key_make(const void *p)
+{
+	const struct bgp_nhc *nhc = p;
+	uint32_t key = 0;
+
+	key = jhash(&nhc->afi, sizeof(nhc->afi), key);
+	key = jhash(&nhc->safi, sizeof(nhc->safi), key);
+	key = jhash(&nhc->nh_length, sizeof(nhc->nh_length), key);
+	key = jhash(&nhc->nh_ipv4, IPV4_MAX_BYTELEN, key);
+	key = jhash(&nhc->nh_ipv6, IPV6_MAX_BYTELEN, key);
+	key = jhash(&nhc->tlvs_length, sizeof(nhc->tlvs_length), key);
+
+	if (nhc->tlvs)
+		key = jhash(nhc->tlvs, nhc->tlvs_length, key);
+
+	return key;
+}
+
+static bool bgp_nhc_hash_cmp(const void *p1, const void *p2)
+{
+	const struct bgp_nhc *nhc1 = p1;
+	const struct bgp_nhc *nhc2 = p2;
+
+	if (nhc1 == nhc2)
+		return true;
+
+	if (!nhc1 || !nhc2)
```

Comment:
I don't think it's possible for both to be null.   What code path are you seeing?

Reply:
It's checked if any of them is null or what do you mean?